src/draw_partial.jl

```diff
@@ -0,0 +1,385 @@
+import Cassette: @context, prehook, overdub
+@context ctx_strokelength
+@context ctx_partial
+
+cur_len_perim = [0.0] #list of length of every stroke
+cur_len_partial = 0.0  # current length for _draw_partial
+target_len_partial = 0.0 # the target length for _draw_partial
+draw_state = true #draw state, functions turn this false 
+#to stop furthur drawing from happening
+
+"""A slightly modified version of pathtopoly ,
+by default pathtopoly() will dispatch to Luxors pathtopoly,
+however passing any symbol will dispatch to the following
+function. Main difference is a new move command starts
+a new subpath, we also return if each subpath is closed or
+open in an Array{Bool},
+"""
+function Luxor.pathtopoly(co_state::Symbol)
```

Comment:
The function name and parameters should be the first line of each docstring. Feel free to check others to see the general style we use.